Add Test PR Quick Start guide (#16)
ansible/community-docs: test_pr_quick_start_guide.rst

It can be useful, particularly, for reviewers and issue authors when they want to check if a PR fixes the reported bug.

.. contents:: Topics

Prepare your environment
========================

We assume that you use Linux as a work environment (you can use a virtual machine as well) and have ``git`` installed.

1. `Install <https://docs.ansible.com/ansible/devel/installation_guide/intro_installation.html>`_ Ansible or ansible-core.

2. Create the following directories in your home directory:

.. code:: bash

  mkdir -p ~/ansible_collections/NAMESPACE/COLLECTION_NAME

For example, if the collection is ``community.general``, it will be:

.. code:: bash

  mkdir -p ~/ansible_collections/community/general

If the collection is ``ansible.posix``, it will be:

.. code:: bash

  mkdir -p ~/ansible_collections/ansible/posix

3. Clone the forked repository from the author profile to the created path:

.. code:: bash

  git clone https://github.com/AUTHOR_ACC/COLLECTION_REPO.git ~/ansible_collections/NAMESPACE/COLLECTION_NAME

4. Go to the cloned repository.

.. code:: bash

  cd ~/ansible_collections/NAMESPACE/COLLECTION_NAME

> 5. Checkout to the PR branch (it can be retrieved from the PR's page):

Author: I'm not sure if this is right or not, checkout to the branch, or checkout the branch.
I think the latter is correct, please put a comment if this change is wrong.
```suggestion
5. Checkout the PR branch (it can be retrieved from the PR's page):
```

.. code:: bash

  git checkout pr_branch

Test the Pull Request
=====================

6. Include `~/ansible-collections` in `COLLECTIONS_PATHS`.

Refer to the `Ansible documentation <https://docs.ansible.com/ansible/devel/reference_appendices/config.html#collections-paths>`_ for details.

> 7. Run your playbook again this time using the changed PR.

Author: ```suggestion
7. Run your playbook again this time using the PR branch.
```

************************
How to test a PR locally
************************

This guide describes all steps needed to test another author's pull request (hereinafter ``PR``).

It can be useful, particularly, for reviewers and issue authors when they want to check if a PR fixes the reported bug.

.. contents:: Topics

Prepare your environment
========================

We assume that you use Linux as a work environment (you can use a virtual machine as well) and have ``git`` installed.

> 1. If possible, make sure that you have installed and started ``docker``. While you can also run tests without docker, this makes it a lot easier since you do not have to install the precise requirements, and tests are running properly isolated and in the exact same environments as in CI. You often can also use ``podman`` with the ``docker`` executable shim, so if you have that you probably do not need to install ``docker``. The latest ansible-core development version also supports the ``podman`` CLI program.

Reviewer: ```suggestion
```
Why keep this if nothing in these instructions mentions `docker` or `podman`?

Author: Right, PR author should fix the module/tests to pass the CI, tester can see CI results there too, there is no point in running `ansible-test` when you want to **just test** a PR.

*****************************
Quick-start development guide
*****************************

This guide describes all steps needed to test another author's pull request (hereinafter ``PR``).

It can be useful, particularly, for reviewers and issue authors when they want to check if a PR fixes the reported bug.

.. contents:: Topics

Prepare your environment
========================

We assume that you use Linux as a work environment (you can use a virtual machine as well) and have ``git`` installed.

> 1. If possible, make sure that you have installed and started ``docker``. While you can also run tests without docker, this makes it a lot easier since you do not have to install the precise requirements, and tests are running properly isolated and in the exact same environments as in CI. You often can also use ``podman`` with the ``docker`` executable shim, so if you have that you probably do not need to install ``docker``.

Reviewer: ```suggestion
1. If possible, make sure that you have installed and started ``docker``. While you can also run tests without docker, this makes it a lot easier since you do not have to install the precise requirements, and tests are running properly isolated and in the exact same environments as in CI. You often can also use ``podman`` with the ``docker`` executable shim, so if you have that you probably do not need to install ``docker``. The latest ansible-core development version also supports the ``podman`` CLI program.
```

Reviewer 2: @aminvakil feel free to adjust the create pr quick start guide similarly

Author: Added to `create_pr_quick_start` in https://github.com/ansible/community-docs/pull/16/commits/40200566a4c573e71886dda715759e853b989abc.

2. `Install <https://docs.ansible.com/ansible/devel/installation_guide/intro_installation.html>`_ Ansible or ansible-core.

3. Create the following directories in your home directory:

.. code:: bash

  mkdir -p ~/ansible_collections/NAMESPACE/COLLECTION_NAME

For example, if the collection is ``community.general``, it will be:

.. code:: bash

  mkdir -p ~/ansible_collections/community/general

If the collection is ``ansible.posix``, it will be:

.. code:: bash

  mkdir -p ~/ansible_collections/ansible/posix

> 4. Fork the collection repository through the GitHub web interface.

Reviewer: This step is not needed, since below you clone the author's repository instead of your own fork.

5. Clone the forked repository from the author profile to the created path:

.. code:: bash

  git clone https://github.com/AUTHOR_ACC/COLLECTION_REPO.git ~/ansible_collections/NAMESPACE/COLLECTION_NAME

6. Go to the cloned repository.

.. code:: bash

  cd ~/ansible_collections/NAMESPACE/COLLECTION_NAME

7. Checkout to the PR branch (it can be retrieved from the PR's page):

.. code:: bash

  git checkout pr_branch

Test the Pull Request
=====================

> 8. Include `~/ansible-collections` in COLLECTIONS_PATHS.

Reviewer: ```suggestion
8. Include `~/ansible-collections` in `COLLECTIONS_PATHS`.
```

Refer to the `Ansible documentation <https://docs.ansible.com/ansible/devel/reference_appendices/config.html#collections-paths>`_ for details.

9. Run your playbook again this time using the changed PR.

Give Feedback
=============

> 12. You can now give feedback on the pull request or the linked issue(s).

Reviewer: The numbering needs to be adjusted :)

*****************************

> Quick-start development guide

Reviewer 2: ```suggestion
Quick-start development guide
```
I tried to figure out how we could change this title but failed. Any ideas?

Author: Test Pull Request locally guide?
Quick-start local PR testing guide?
Neither one looks good to me :(

Reviewer 2: maybe
```
How to test another's PR locally
```
? (to highlight the case)

Reviewer: How about `How to test a PR locally` instead? Or write out `another's` as `another person's`?

Reviewer 2: sounds good to me:)

Reviewer 2: both

Author: @Andersson007 @felixfontein This and other conversations have been fixed in https://github.com/ansible/community-docs/pull/16/commits/e7d32b730e343d177fa1d2444bf427c0c1747202,
can you please take another look?

*****************************

> This guide describes all steps needed to test another author's pull request.

Reviewer 2: ```suggestion
This guide describes all steps needed to test another author's pull request (hereinafter ``PR``).
```

> This guide describes all steps needed to test another author's pull request (hereinafter ``PR``).

Reviewer 2: ```suggestion
This guide describes all steps needed to test another author's pull request (hereinafter ``PR``).

It can be useful, for example, for reviewers and issue authors when they want to check if a PR fixes the reported bug.
```

Author: I've done a minor change in your sentence in https://github.com/ansible/community-docs/pull/16/commits/78854b70fd32d02aca4688f443dbda61386d2b27.

.. contents:: Topics

Prepare your environment
========================

We assume that you use Linux as a work environment (you can use a virtual machine as well) and have ``git`` installed.

1. If possible, make sure that you have installed and started ``docker``. While you can also run tests without docker, this makes it a lot easier since you do not have to install the precise requirements, and tests are running properly isolated and in the exact same environments as in CI. You often can also use ``podman`` with the ``docker`` executable shim, so if you have that you probably do not need to install ``docker``.

2. `Install <https://docs.ansible.com/ansible/devel/installation_guide/intro_installation.html>`_ Ansible or ansible-core.

3. Create the following directories in your home directory:

.. code:: bash

  mkdir -p ~/ansible_collections/NAMESPACE/COLLECTION_NAME

For example, if the collection is ``community.general``, it will be:

.. code:: bash

  mkdir -p ~/ansible_collections/community/general

> If the collection is ``ansible.posix``, it will be:

Reviewer 2: ```suggestion
If the collection is ``ansible.posix``, it will be:
```
it's a good idea to provide this additional example. Maybe add the same to `create_pr_quick_start_guide.rst`?

Author: I've copied it from `create_pr_quick_start_guide.rst`, so it's already there. Or do you mean adding this somewhere else?

Reviewer 2: ah, no then:) I forgot that it's there

.. code:: bash

  mkdir -p ~/ansible_collections/ansible/posix

4. Fork the collection repository through the GitHub web interface.

5. Clone the forked repository from the author profile to the created path:

.. code:: bash

  git clone https://github.com/AUTHOR_ACC/COLLECTION_REPO.git ~/ansible_collections/NAMESPACE/COLLECTION_NAME

6. Go to the cloned repository.

.. code:: bash

  cd ~/ansible_collections/NAMESPACE/COLLECTION_NAME

> 7. Checkout to the PR branch (it can be retrieved from PR page):

Reviewer 2: ```suggestion
7. Checkout to the PR branch (it can be retrieved from the PR's page):
```

.. code:: bash

  git checkout pr_branch

> Test the PR
> =================

Reviewer 2: ```suggestion
Test the pull request
=====================
```
I'm OK with using ``PR`` in the paragraphs. I'd avoid of using the abbreviation in the paragraph titles. Though it's just a suggestion, feel free to ignore.

8. Include `~/ansible-collections` in COLLECTIONS_PATHS.

> More info how to add that can be found in https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths.

Reviewer 2: ```suggestion
Refer to the `Ansible documentation <https://docs.ansible.com/ansible/devel/reference_appendices/config.html#collections-paths>`_ for details.
```

9. Run your playbook again this time using the changed PR.

> 10. Run sanity tests:
> 
> .. code:: bash
> 
>   ansible-test sanity path/to/changed_file.py --docker -v
> 
> If they failed, look at the output carefully - it is usually very informative and helps to identify a problem line quickly.
> Sanity failings usually relate to wrong code and documentation formatting.
> 
> 11. Run integration tests:
> 
> .. code:: bash
> 
>   ansible-test integration name_of_test_subdirectory --docker -v
> 
> For example, if the tests files you changed are stored in ``tests/integration/targets/test_mysql_user/``, the command will be:
> 
> .. code:: bash
> 
>   ansible-test integration test_mysql_user --docker -v
> 
> You can use the ``-vv`` or ``-vvv`` argument, if you need more detailed output.
> 

Reviewer 2: ```suggestion
```
I think we can exclude this. What do you think?

Author: I agree, these have been already run in CI.

Give Feedback
=============

> 12. You can now give feedback on Pull Request or the linked issue(s).

Reviewer 2: ```suggestion
12. You can now give feedback on the pull request or the linked issue(s).
```
